repos/fdbt-site/src/pages/csvZoneUpload.tsx

```diff
@@ -72,17 +72,28 @@ const CsvZoneUpload = ({
     const uniqueServiceLists =
         serviceList.filter((item) => (seen.includes(item.lineId) ? false : seen.push(item.lineId))) ?? [];
 
-    const isDefaultChecked = serviceList.every((service) => service.checked);
     const [getButtonText, updateButtonText] = useState(buttonText);
-    const [isCheckedAll, updateChecked] = useState(isDefaultChecked);
+    const [checkedServices, updateChecked] = useState(uniqueServiceLists.filter((service) => service.checked));
 
     const toggleAllServices = (): void => {
-        if (!isCheckedAll) {
+        if (checkedServices.length !== uniqueServiceLists.length) {
             updateButtonText('Unselect All Services');
+            updateChecked(uniqueServiceLists);
         } else {
             updateButtonText('Select All Services');
+            updateChecked([]);
+        }
+    };
```

Comment:
If i click select all, then unselect some services, then click 'unselect all', it selects the ones I unselected.